ports/android/README.md

```diff
@@ -0,0 +1,149 @@
+# Android port
+
+Building for Android 9 (codename Pie), or API level 28. Earlier versions of
+Android may be supported at some point, but that would require implementing
+replacements for `pthread_attr_setinheritsched` (introduced in level 28) and
+`getifaddrs` (introduced in level 24), which is outside the scope of this
+document.
+
+> Linux, specifically Fedora 31, was used to create this guide. The steps to
+> build Eclipse Cyclone DDS should be more-or-less the same on other
+> platforms, the steps to setup the emulator are probably not.
+
+
+## Installing the Android SDK
+
+[1]: https://developer.android.com/studio
+[2]: https://stackoverflow.com/questions/43923996/adb-root-is-not-working-on-emulator-cannot-run-as-root-in-production-builds
+
+ * Download [Android Studio](1) and extract the archive.
+
+ * Run `android-studio/bin/studio.sh` to launch the *Android Studio Setup
+   Wizard* and install the *Android SDK*. Select the *Android Virtual Device*
+   option to install the emulator.
+
+ * Click *Configure* from the *Android Studio* welcome screen and select
+   *SDK Manager*. Switch to the *SDK Tools* tab, select *NDK (Side by side)*,
+   click *Ok* and wait for the **Android Native Development Kit (NDK)** to
+   finish downloading.
+
+> *Configure* is available from the *Welcome to Android Studio* window, but
+> maybe hidden because it is not appropriatly sized. Resize the window so
+> *Configure* and *Get Help* are shown.
+
+ * Click *Configure* from the *Android Studio* welcome screen and select
+   *AVD Manager*. Click *Create Virtual Device*, select the *Phone* category
+   and choose the *Pixel XL* device definition. Click *Next* and hit
+   *Download* next to *Q*. *Accept* the *License Agreement* and click *Next*
+   to download the emulator image. When the download is finished, select *Q*,
+   click *Next*, then *Finish* to create the *Android Virtual Device (AVD)*.
+
+> As can be read from [this StackOverflow post](2), it is important to NOT
+> select a *Google Play* image, as these images do not allow you to gain
+> root privileges.
+
+
+## Building Eclipse Cyclone DDS
+
+[3]: https://developer.android.com/ndk/guides/cmake
+[4]: https://developer.android.com/ndk/guides/cmake#variables
+
+The [Android NDK supports CMake](3) via a toolchain file. Build parameters
+such as ABI can specified on the command line. For the complete list of
+supported variables, consult the [Toolchain Arguments](4) section.
+
+```
+$ cd cyclonedds
+$ mkdir build.android
+$ cd build.android
+$ cmake -DCMAKE_TOOLCHAIN_FILE=<path/to/Android/Sdk>/ndk/<version>/build/cmake/android.toolchain.cmake \
+        -DANDROID_ABI=x86 -DANDROID_NATIVE_API_LEVEL=28 -DBUILD_SHARED_LIBS=Off ..
+$ cmake --build .
+```
+
+> Android supports shared libraries. `BUILD_SHARED_LIBS` is disabled solely
+> for demonstration purposes.
+
+
+## Setting up a networking bridge
```

Comment:
remove network setup or move to fyi section at end

Reply:
The network setup is actually the most interesting part because it contains the bare minimum of steps to get things working. Makes debugging etc a lot easier, so a people likely want to have this information once they get going. Apart from that it shows that it'll communicate over the network, not just `lo`. I can move it to the end and add a local test step in between.